### PR TITLE
fix: set_response_header breaking openapi doc rendering

### LINKED ISSUE #976: `app.set_response_header`will break Openapi doc rendering
### Bug Description

`app.set_response_header("Content-Type", "application/json")` will add additional content type to `/docs` endpoint, which can disable browsers to render the html content.

This can be fixed by prioritizing the content type in response (around https://github.com/sparckles/Robyn/blob/main/src/types/headers.rs#L161)

### Steps to Reproduce

```py
from pydantic import BaseModel
from robyn import Request, Robyn

app = Robyn(__file__)

app.set_response_header("Content-Type", "application/json")


class Response(BaseModel):
    text: str


@app.get("/")
async def h(request: Request):
    return Response(text="hello")


app.start(port=8080)
```

Then you'll see:

image

```
$ curl -v http://localhost:8080/docs/
...
> Host: localhost:8080
> User-Agent: curl/8.7.1
> Accept: */*
> 
* Request completely sent off
< HTTP/1.1 200 OK
< content-length: 938
< content-type: text/html
< content-type: application/json
< date: Sun, 06 Oct 2024 10:56:04 GMT
...
```

### Your operating system

None

### Your Python version (`python --version`)

None

### Your Robyn version

latest

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by VishnuSanal**
## Description

This PR fixes #976

## Summary

This PR fixes `set_response_header` breaking openapi doc rendering

## PR Checklist

Please ensure that:

- [x] The PR contains a descriptive title
- [x] The PR contains a descriptive summary of the changes
- [x] You build and test your changes before submitting a PR.
- [x] You have added relevant documentation
- [x] You have added relevant tests. We prefer integration tests wherever possible

## Pre-Commit Instructions:

- [x] Ensure that you have run the [pre-commit hooks](https://github.com/sparckles/robyn#%EF%B8%8F-to-develop-locally) on your PR.

### CODE REVIEW

**Comment by VishnuSanal** [comment]
@kigawas can you have a look here?

File: src/server.rs [hunk 1]
```diff
@@ -452,6 +452,12 @@ async fn index(
 
     response.headers.extend(&global_response_headers);
 
+    if req.uri().path().eq("/docs") {
+        response
+            .headers
+            .append("Content-Type".to_string(), "text/html".to_string());
+    }
+
```
Comment (sansyrox) [inline src/server.rs:460]: @VishnuSanal , how does this work in case of a const request?

Comment (VishnuSanal) [reply src/server.rs:460]: Didn't get you, `/docs` was a const request, BTW.

Comment (sansyrox) [reply src/server.rs:460]: Exactly. So, how are handling that case?

File: src/server.rs [hunk 2]
```diff
@@ -452,6 +452,12 @@ async fn index(
 
     response.headers.extend(&global_response_headers);
 
+    if req.uri().path().eq("/docs") {
+        response
```
Comment (kigawas) [inline src/server.rs:456]: I'd expect a more fundamental fix of response headers rather than checking path

File: src/server.rs (hunk 1 above)
Comment (VishnuSanal) [reply src/server.rs:460]: I still didn't get your question, I am afraid. `/docs` was a const request & this works perfectly fine. plus, nobody can alter the docs page behavior too.

File: src/server.rs (hunk 2 above)
Comment (kigawas) [reply src/server.rs:456]: IMO there should be only one `Content-Type` header in the response, maybe it's better to 
- add some content-type resolution logic to prioritize `text/html`; or
- add an `exclude_paths: list[str]` argument to `set_response_header`; or
- add a `custom_response_header: dict[str, str]` argument to`add_route` 

Any of these is better than hard coding "/docs"

Comment (VishnuSanal) [reply src/server.rs]: @kigawas @sansyrox PTAL

File: robyn/__init__.py [hunk 3]
```diff
@@ -200,6 +200,9 @@ def set_request_header(self, key: str, value: str) -> None:
     def set_response_header(self, key: str, value: str) -> None:
         self.response_headers.set(key, value)
 
+    def exclude_response_headers(self, exclude_paths: Optional[List[str]]):
+        self.response_headers.set_exclude_paths(exclude_paths)
```
Comment (sansyrox) [inline robyn/__init__.py:204]: @VishnuSanal , should this be a public function?

File: robyn/robyn.pyi [hunk 4]
```diff
@@ -177,6 +177,8 @@ class QueryParams:
         pass
 
 class Headers:
+    exclude_paths: Optional[list[str]] = None
```
Comment (sansyrox) [inline robyn/robyn.pyi:180]: Why does the `Headers` class need to have an `exclude_paths` function?

Comment (VishnuSanal) [reply robyn/robyn.pyi:180]: `exclude_paths` is not a function, it is a list of string -- to track the excluded paths, to be passed to rust.

File: robyn/__init__.py (hunk 3 above)
Comment (VishnuSanal) [reply robyn/__init__.py:204]: yep, why not? what if the user wants to enforce it to some pages themselves?

Comment (sansyrox) [reply robyn/__init__.py:204]: Then we'd need to add documentation :)

File: docs_src/src/pages/documentation/api_reference/getting_started.mdx [hunk 5]
```diff
@@ -491,12 +491,11 @@ Either, by using the `headers` field in the `Response` object:
 ### Global Response Headers
 
 <Row>
-<Col>
-Or setting the Headers globally *per* router.
-</Col>
-<Col>
 
+{' '}
```
Comment (sansyrox) [inline docs_src/src/pages/documentation/api_reference/getting_started.mdx:495]: extra

File: robyn/__init__.py [hunk 6]
```diff
@@ -200,6 +200,9 @@ def set_request_header(self, key: str, value: str) -> None:
     def set_response_header(self, key: str, value: str) -> None:
         self.response_headers.set(key, value)
 
+    def exclude_response_headers(self, exclude_paths: Optional[List[str]]):
```
Comment (sansyrox) [inline robyn/__init__.py:203]: can you please add a docstring here

File: robyn/__init__.py [hunk 7]
```diff
@@ -239,6 +242,7 @@ def _add_openapi_routes(self, auth_required: bool = False):
             is_const=True,
             auth_required=auth_required,
         )
+        self.exclude_response_headers(["/docs"])
```
Comment (sansyrox) [inline robyn/__init__.py:245]: what do you think about `/openapi.json` ?

File: robyn/robyn.pyi (hunk 4 above)
Comment (sansyrox) [reply robyn/robyn.pyi:180]: I am not sure if we should also have this information here. Let me think more

File: robyn/__init__.py (hunk 7 above)
Comment (VishnuSanal) [reply robyn/__init__.py:245]: should I add it? I didn't think of a case that should be added ...

- update: just discovered all of the types! ref: https://stackoverflow.com/a/48704300/9652621, will add!

File: robyn/robyn.pyi (hunk 4 above)
Comment (sansyrox) [inline robyn/robyn.pyi:180]: Yes, headers should not have an `exclude_paths` method. This is not a great design. You need to figure out a way to store this somewhere else

Comment (VishnuSanal) [reply robyn/robyn.pyi:180]: @sansyrox how about adding it to the `Server` struct? just like `global_request_headers`, & `global_response_headers`?

Comment (sansyrox) [reply robyn/robyn.pyi:180]: @VishnuSanal , that should be a better place. But the data structure should not be super general.

File: src/server.rs [hunk 8]
```diff
@@ -414,11 +424,11 @@ async fn index(
     payload: web::Payload,
     const_router: web::Data<Arc<ConstRouter>>,
     middleware_router: web::Data<Arc<MiddlewareRouter>>,
-    global_request_headers: web::Data<Arc<Headers>>,
-    global_response_headers: web::Data<Arc<Headers>>,
```
Comment (sansyrox) [inline src/server.rs:418]: please don't change this

File: src/server.rs [hunk 9]
```diff
@@ -414,11 +424,11 @@ async fn index(
...
+    global_headers: (web::Data<Arc<Headers>>, web::Data<Arc<Headers>>),
+    exclude_paths: web::Data<Option<Vec<String>>>,
```
Comment (sansyrox) [inline src/server.rs:428]: `exclude_paths` is not descriptive enough. Could you make it more descriptive?

File: robyn/__init__.py [hunk 10]
```diff
@@ -200,6 +201,13 @@ def set_request_header(self, key: str, value: str) -> None:
     def set_response_header(self, key: str, value: str) -> None:
         self.response_headers.set(key, value)
 
+    def exclude_response_headers(self, exclude_paths: Optional[List[str]]):
```
Comment (sansyrox) [inline robyn/__init__.py:204]: this function name implies we are passing headers as a parameter. We should come up with a better name.

File: docs_src/src/pages/documentation/api_reference/getting_started.mdx [hunk 11]
```diff
@@ -594,7 +590,25 @@ Or setting the Headers globally *per* router.
     app.set_response_header("content-type", "application/json")
     ```
     </CodeGroup>
+
   </Col>
+  
+<Col>
+You can disable some routes from the application of response headers with `exclude_response_headers`.
```
Comment (sansyrox) [inline docs_src/src/pages/documentation/api_reference/getting_started.mdx:597]: should reword this.

File: src/types/headers.rs [hunk 12]
```diff
@@ -158,6 +158,10 @@ impl Headers {
         self.headers.remove(&key.to_lowercase());
     }
 
+    pub fn remove_all(&mut self) {
+        self.headers.clear();
+    }
```
Comment (sansyrox) [inline src/types/headers.rs:163]: is it needed?

File: src/server.rs (hunk 8 above)
Comment (VishnuSanal) [reply src/server.rs:418]: this was to fix ci: cargo clippy warning https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments

any other suggestions to do so?

File: src/types/headers.rs (hunk 12 above)
Comment (VishnuSanal) [reply src/types/headers.rs:163]: yes. 🤷🏿  it is used on server.rs#497

File: docs_src/src/pages/documentation/api_reference/getting_started.mdx [hunk 13]
```diff
@@ -594,7 +590,25 @@ Or setting the Headers globally *per* router.
...
   </Col>
+  
+<Col>
+To prevent the headers from getting applied to certain endpoints, you can use the `response_headers_exclude` function.
+</Col>
+<Col>
+    <CodeGroup title="Request" tag="GET" label="/hello_world">
+
+    ```python {{ title: 'untyped' }}
+    app.response_headers_exclude(["/login", "/signup"])
```
Comment (sansyrox) [inline docs_src/src/pages/documentation/api_reference/getting_started.mdx:603]: could you rename it to `exclude_response_headers_for`

File: robyn/__init__.py [hunk 14]
```diff
@@ -200,6 +201,13 @@ def set_request_header(self, key: str, value: str) -> None:
     def set_response_header(self, key: str, value: str) -> None:
         self.response_headers.set(key, value)
 
+    def response_headers_exclude(self, exclude_paths: Optional[List[str]]):
+        """
+        To disable certain routes from the application of response headers
```
Comment (sansyrox) [inline robyn/__init__.py:206]:
```suggestion
        To exclude response headers from certain routes
```

File: robyn/processpool.py [hunk 15]
```diff
@@ -27,6 +27,7 @@ def run_processes(
     workers: int,
     processes: int,
     response_headers: Headers,
+    header_exclude_paths: Optional[List[str]],
```
Comment (sansyrox) [inline robyn/processpool.py:30]: response_headers*

File: robyn/processpool.py [hunk 16]
```diff
@@ -43,6 +44,7 @@ def run_processes(
         workers,
         processes,
         response_headers,
+        header_exclude_paths,
```
Comment (sansyrox) [inline robyn/processpool.py:47]: same

File: robyn/processpool.py [hunk 17]
```diff
@@ -76,6 +78,7 @@ def init_processpool(
     workers: int,
     processes: int,
     response_headers: Headers,
+    header_exclude_paths: Optional[List[str]],
```
Comment (sansyrox) [inline robyn/processpool.py:81]: same

File: robyn/processpool.py [hunk 18]
```diff
@@ -144,6 +149,7 @@ def spawn_process(
     socket: SocketHeld,
     workers: int,
     response_headers: Headers,
+    header_exclude_paths: Optional[List[str]],
```
Comment (sansyrox) [inline robyn/processpool.py:152]: same

File: robyn/processpool.py [hunk 19]
```diff
@@ -173,6 +179,8 @@ def spawn_process(
 
     server.apply_response_headers(response_headers)
 
+    server.set_response_header_exclude_paths(header_exclude_paths)
```
Comment (sansyrox) [inline robyn/processpool.py:182]: same

File: src/server.rs (hunk 8 above)
Comment (sansyrox) [reply src/server.rs:418]: I meant the name.

`global_request_headers` and `global_response_headers`

File: src/server.rs [hunk 20]
```diff
@@ -414,11 +427,11 @@ async fn index(
...
     const_router: web::Data<Arc<ConstRouter>>,
     middleware_router: web::Data<Arc<MiddlewareRouter>>,
-    global_request_headers: web::Data<Arc<Headers>>,
-    global_response_headers: web::Data<Arc<Headers>>,
+    global_headers: (web::Data<Arc<Headers>>, web::Data<Arc<Headers>>),
+    response_header_exclude_paths: web::Data<Option<Vec<String>>>,
     req: HttpRequest,
 ) -> impl Responder {
-    let mut request = Request::from_actix_request(&req, payload, &global_request_headers).await;
+    let mut request = Request::from_actix_request(&req, payload, &global_headers.0).await;
```
Comment (sansyrox) [inline src/server.rs:434]: I am not a big fan of a tuple style access here. please keep it to global_request_headers

File: src/server.rs [hunk 21]
```diff
@@ -478,7 +491,16 @@ async fn index(
 
     debug!("OG Response : {:?}", response);
 
-    response.headers.extend(&global_response_headers);
+    response.headers.extend(&global_headers.1);
```
Comment (sansyrox) [inline src/server.rs:494]: same here `global_response_headers`

File: src/types/headers.rs [hunk 22]
```diff
@@ -158,6 +158,10 @@ impl Headers {
         self.headers.remove(&key.to_lowercase());
     }
 
+    pub fn remove_all(&mut self) {
```
Comment (sansyrox) [inline src/types/headers.rs:161]:
```suggestion
    pub fn clear_all(&mut self) {
```

File: src/server.rs [hunk 23]
```diff
@@ -478,7 +491,16 @@ async fn index(
...
+
+    match &response_header_exclude_paths.get_ref() {
+        None => {}
+        Some(exclude_paths) => {
+            if exclude_paths.contains(&req.uri().path().to_owned()) {
+                response.headers.remove_all();
```
Comment (sansyrox) [inline src/server.rs:500]:
```suggestion
                response.headers.clear();
```

File: src/server.rs (hunk 20 above)
Comment (VishnuSanal) [reply src/server.rs:434]: this was to fix ci: cargo clippy warning https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments

any other suggestions to do so?

**Comment by VishnuSanal** [comment]
- disable clippy warning for index()

File: robyn/__init__.py [hunk 24]
```diff
@@ -70,6 +70,7 @@ def __init__(
         self.web_socket_router = WebSocketRouter()
         self.request_headers: Headers = Headers({})
         self.response_headers: Headers = Headers({})
+        self.header_exclude_paths: Optional[List[str]] = None
```
Comment (sansyrox) [inline robyn/__init__.py:73]: rename to `excluded_response_header_paths`

File: robyn/__init__.py [hunk 25]
```diff
@@ -200,6 +201,13 @@ def set_request_header(self, key: str, value: str) -> None:
     def set_response_header(self, key: str, value: str) -> None:
         self.response_headers.set(key, value)
 
+    def exclude_response_headers_for(self, exclude_paths: Optional[List[str]]):
```
Comment (sansyrox) [inline robyn/__init__.py:204]: `excluded_paths`
```suggestion
    def exclude_response_headers_for(self, excluded_paths: Optional[List[str]]):
```

File: robyn/processpool.py [hunk 26]
```diff
@@ -27,6 +27,7 @@ def run_processes(
     workers: int,
     processes: int,
     response_headers: Headers,
+    response_header_exclude_paths: Optional[List[str]],
```
Comment (sansyrox) [inline robyn/processpool.py:30]: excluded_response_header_paths

File: robyn/processpool.py [hunk 27]
```diff
@@ -43,6 +44,7 @@ def run_processes(
         workers,
         processes,
         response_headers,
+        response_header_exclude_paths,
```
Comment (sansyrox) [inline robyn/processpool.py:47]: excluded_response_header_paths

File: robyn/processpool.py [hunk 28]
```diff
@@ -76,6 +78,7 @@ def init_processpool(
     workers: int,
     processes: int,
     response_headers: Headers,
+    response_header_exclude_paths: Optional[List[str]],
```
Comment (sansyrox) [inline robyn/processpool.py:81]: excluded_response_header_paths

File: robyn/processpool.py [hunk 29]
```diff
@@ -90,6 +93,7 @@ def init_processpool(
             socket,
             workers,
             response_headers,
+            response_header_exclude_paths,
```
Comment (sansyrox) [inline robyn/processpool.py:96]:
```suggestion
            excluded_response_header_paths,
```

File: robyn/robyn.pyi [hunk 30]
```diff
@@ -327,6 +327,8 @@ class Server:
         pass
     def apply_response_headers(self, headers: Headers) -> None:
         pass
+    def set_response_header_exclude_paths(self, response_header_exclude_paths: Optional[list[str]] = None):
```
Comment (sansyrox) [inline robyn/robyn.pyi:330]:
```suggestion
    def set_response_header_exclude_paths(self, excluded_response_header_paths: Optional[list[str]] = None):
```

File: src/server.rs [hunk 31]
```diff
@@ -56,6 +56,7 @@ pub struct Server {
     directories: Arc<RwLock<Vec<Directory>>>,
     startup_handler: Option<Arc<FunctionInfo>>,
     shutdown_handler: Option<Arc<FunctionInfo>>,
+    response_header_exclude_paths: Option<Vec<String>>,
```
Comment (sansyrox) [inline src/server.rs:59]:
```suggestion
    excluded_response_header_paths: Option<Vec<String>>,
```

File: src/server.rs [hunk 32]
```diff
@@ -72,6 +73,7 @@ impl Server {
             directories: Arc::new(RwLock::new(Vec::new())),
             startup_handler: None,
             shutdown_handler: None,
+            response_header_exclude_paths: None,
```
Comment (sansyrox) [inline src/server.rs:76]: same

File: src/server.rs [hunk 33]
```diff
@@ -478,7 +491,16 @@ async fn index(
 
     debug!("OG Response : {:?}", response);
 
-    response.headers.extend(&global_response_headers);
+    response.headers.extend(&global_request_response_headers.1);
+
+    match &response_header_exclude_paths.get_ref() {
```
Comment (sansyrox) [inline src/server.rs:496]:
```suggestion
    match &excluded_response_header_paths.get_ref() {
```

File: src/server.rs [hunk 34]
```diff
@@ -478,7 +491,16 @@ async fn index(
...
+        None => {}
+        Some(exclude_paths) => {
```
Comment (sansyrox) [inline src/server.rs:498]:
```suggestion
        Some(exclude_paths) => {
```
something else?

File: src/server.rs [hunk 35]
```diff
@@ -414,11 +427,11 @@ async fn index(
...
     const_router: web::Data<Arc<ConstRouter>>,
     middleware_router: web::Data<Arc<MiddlewareRouter>>,
-    global_request_headers: web::Data<Arc<Headers>>,
-    global_response_headers: web::Data<Arc<Headers>>,
+    global_request_response_headers: (web::Data<Arc<Headers>>, web::Data<Arc<Headers>>),
+    response_header_exclude_paths: web::Data<Option<Vec<String>>>,
     req: HttpRequest,
 ) -> impl Responder {
-    let mut request = Request::from_actix_request(&req, payload, &global_request_headers).await;
+    let mut request = Request::from_actix_request(&req, payload, &global_request_response_headers.0).await;
```
Comment (sansyrox) [inline src/server.rs:434]: no tuples!

File: src/server.rs [hunk 36]
```diff
@@ -478,7 +491,16 @@ async fn index(
 
     debug!("OG Response : {:?}", response);
 
-    response.headers.extend(&global_response_headers);
+    response.headers.extend(&global_request_response_headers.1);
```
Comment (sansyrox) [inline src/server.rs:494]: same here

File: src/server.rs [hunk 37]
```diff
@@ -298,6 +304,13 @@ impl Server {
         self.global_response_headers = Arc::new(headers.clone());
     }
 
+    pub fn set_response_header_exclude_paths(
+        &mut self,
+        response_header_exclude_paths: Option<Vec<String>>,
```
Comment (sansyrox) [inline src/server.rs:309]:
```suggestion
        excluded_response_header_paths: Option<Vec<String>>,
```

File: src/server.rs (hunk 34 above)
Comment (VishnuSanal) [reply src/server.rs:498]: what?

Comment (sansyrox) [reply src/server.rs:501]: You should name `exclude_paths` differently @VishnuSanal

File: src/server.rs [hunk 38]
```diff
@@ -298,6 +305,13 @@ impl Server {
         self.global_response_headers = Arc::new(headers.clone());
     }
 
+    pub fn set_response_header_exclude_paths(
```
Comment (sansyrox) [inline src/server.rs:308]:
```suggestion
    pub fn set_response_headers_exclude_paths(
```

File: src/server.rs [hunk 39]
```diff
@@ -298,6 +305,13 @@ impl Server {
...
+        &mut self,
+        excluded_response_header_paths: Option<Vec<String>>,
+    ) {
+        self.excluded_response_header_paths = excluded_response_header_paths;
```
Comment (sansyrox) [inline src/server.rs:312]:
```suggestion
        self.excluded_response_headers_paths = excluded_response_headers_paths;
```

File: src/server.rs [hunk 40]
```diff
@@ -409,13 +423,15 @@ impl Default for Server {
...
 /// path, and returns a Future of a Response.
+#[allow(clippy::too_many_arguments)]
 async fn index(
     router: web::Data<Arc<HttpRouter>>,
     payload: web::Payload,
     const_router: web::Data<Arc<ConstRouter>>,
     middleware_router: web::Data<Arc<MiddlewareRouter>>,
     global_request_headers: web::Data<Arc<Headers>>,
     global_response_headers: web::Data<Arc<Headers>>,
+    excluded_response_header_paths: web::Data<Option<Vec<String>>>,
```
Comment (sansyrox) [inline src/server.rs:434]:
```suggestion
    excluded_response_headers_paths: web::Data<Option<Vec<String>>>,
```

File: src/server.rs [hunk 41]
```diff
@@ -480,6 +496,15 @@ async fn index(
 
     response.headers.extend(&global_response_headers);
 
+    match &excluded_response_header_paths.get_ref() {
```
Comment (sansyrox) [inline src/server.rs:499]:
```suggestion
    match &excluded_response_headers_paths.get_ref() {
```

File: src/server.rs [hunk 42]
```diff
@@ -480,6 +496,15 @@ async fn index(
...
+        None => {}
+        Some(excluded_response_header_paths) => {
```
Comment (sansyrox) [inline src/server.rs:501]:
```suggestion
        Some(excluded_response_headers_paths) => {
```

File: src/server.rs [hunk 43]
```diff
@@ -480,6 +496,15 @@ async fn index(
...
+            if excluded_response_header_paths.contains(&req.uri().path().to_owned()) {
```
Comment (sansyrox) [inline src/server.rs:502]:
```suggestion
            if excluded_response_headers_paths.contains(&req.uri().path().to_owned()) {
```

File: src/server.rs [hunk 44]
```diff
@@ -203,6 +209,7 @@ impl Server {
                                         middleware_router,
                                         global_request_headers,
                                         global_response_headers,
+                                        response_header_exclude_paths,
```
Comment (sansyrox) [inline src/server.rs:212]:
```suggestion
                                        response_headers_exclude_paths,
```

File: src/server.rs [hunk 45]
```diff
@@ -194,6 +199,7 @@ impl Server {
                                   payload: web::Payload,
                                   global_request_headers,
                                   global_response_headers,
+                                  response_header_exclude_paths,
```
Comment (sansyrox) [inline src/server.rs:202]:
```suggestion
                                  response_headers_exclude_paths,
```

File: src/server.rs [hunk 46]
```diff
@@ -162,7 +166,8 @@ impl Server {
                         .app_data(web::Data::new(const_router.clone()))
                         .app_data(web::Data::new(middleware_router.clone()))
                         .app_data(web::Data::new(global_request_headers.clone()))
-                        .app_data(web::Data::new(global_response_headers.clone()));
+                        .app_data(web::Data::new(global_response_headers.clone()))
+                        .app_data(web::Data::new(excluded_response_header_paths.clone()));
```
Comment (sansyrox) [inline src/server.rs:170]:
```suggestion
                        .app_data(web::Data::new(excluded_response_headers_paths.clone()));
```

File: src/server.rs [hunk 47]
```diff
@@ -72,6 +73,7 @@ impl Server {
             directories: Arc::new(RwLock::new(Vec::new())),
             startup_handler: None,
             shutdown_handler: None,
+            excluded_response_header_paths: None,
```
Comment (sansyrox) [inline src/server.rs:76]:
```suggestion
            excluded_response_headers_paths: None,
```

File: src/server.rs [hunk 48]
```diff
@@ -56,6 +56,7 @@ pub struct Server {
     directories: Arc<RwLock<Vec<Directory>>>,
     startup_handler: Option<Arc<FunctionInfo>>,
     shutdown_handler: Option<Arc<FunctionInfo>>,
+    excluded_response_header_paths: Option<Vec<String>>,
```
Comment (sansyrox) [inline src/server.rs:59]:
```suggestion
    excluded_response_headers_paths: Option<Vec<String>>,
```

File: robyn/robyn.pyi [hunk 49]
```diff
@@ -327,6 +327,8 @@ class Server:
         pass
     def apply_response_headers(self, headers: Headers) -> None:
         pass
+    def set_response_header_exclude_paths(self, excluded_response_header_paths: Optional[list[str]] = None):
```
Comment (sansyrox) [inline robyn/robyn.pyi:330]:
```suggestion
    def set_response_headers_exclude_paths(self, excluded_response_header_paths: Optional[list[str]] = None):
```

File: robyn/processpool.py [hunk 50]
```diff
@@ -27,6 +27,7 @@ def run_processes(
     workers: int,
     processes: int,
     response_headers: Headers,
+    excluded_response_header_paths: Optional[List[str]],
```
Comment (sansyrox) [inline robyn/processpool.py:30]:
```suggestion
    excluded_response_headers_paths: Optional[List[str]],
```

File: robyn/processpool.py [hunk 51]
```diff
@@ -43,6 +44,7 @@ def run_processes(
         workers,
         processes,
         response_headers,
+        excluded_response_header_paths,
```
Comment (sansyrox) [inline robyn/processpool.py:47]:
```suggestion
        excluded_response_headers_paths,
```

File: robyn/processpool.py [hunk 52]
```diff
@@ -76,6 +78,7 @@ def init_processpool(
     workers: int,
     processes: int,
     response_headers: Headers,
+    excluded_response_header_paths: Optional[List[str]],
```
Comment (sansyrox) [inline robyn/processpool.py:81]:
```suggestion
    excluded_response_headers_paths: Optional[List[str]],
```

File: robyn/processpool.py [hunk 53]
```diff
@@ -90,6 +93,7 @@ def init_processpool(
             socket,
             workers,
             response_headers,
+            excluded_response_header_paths,
```
Comment (sansyrox) [inline robyn/processpool.py:96]:
```suggestion
            excluded_response_headers_paths,
```

File: robyn/processpool.py [hunk 54]
```diff
@@ -109,6 +113,7 @@ def init_processpool(
                 copied_socket,
                 workers,
                 response_headers,
+                excluded_response_header_paths,
```
Comment (sansyrox) [inline robyn/processpool.py:116]:
```suggestion
                excluded_response_headers_paths,
```

File: robyn/processpool.py [hunk 55]
```diff
@@ -144,6 +149,7 @@ def spawn_process(
     socket: SocketHeld,
     workers: int,
     response_headers: Headers,
+    excluded_response_header_paths: Optional[List[str]],
```
Comment (sansyrox) [inline robyn/processpool.py:152]:
```suggestion
    excluded_response_headers_paths: Optional[List[str]],
```

File: robyn/processpool.py [hunk 56]
```diff
@@ -173,6 +179,8 @@ def spawn_process(
 
     server.apply_response_headers(response_headers)
 
+    server.set_response_header_exclude_paths(excluded_response_header_paths)
```
Comment (sansyrox) [inline robyn/processpool.py:182]:
```suggestion
    server.set_response_headers_exclude_paths(excluded_response_header_paths)
```

File: robyn/__init__.py [hunk 57]
```diff
@@ -70,6 +70,7 @@ def __init__(
         self.web_socket_router = WebSocketRouter()
         self.request_headers: Headers = Headers({})
         self.response_headers: Headers = Headers({})
+        self.excluded_response_header_paths: Optional[List[str]] = None
```
Comment (sansyrox) [inline robyn/__init__.py:73]:
```suggestion
        self.excluded_response_headers_paths: Optional[List[str]] = None
```

**Review by sansyrox — COMMENTED** [review]
Hey @VishnuSanal 

I have some follow ups

**Comment by sansyrox** [comment]
Hey @VishnuSanal ,

Can you fetch the latest changes here?

**Review by sansyrox — APPROVED** [review]
LGTM ! 

Good job @VishnuSanal ✨

**Pull request merged**